pkg/monitor/cluster/certificateexpirationstatuses_test.go

```diff
@@ -0,0 +1,235 @@
+package cluster
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+	utiltls "github.com/Azure/ARO-RP/pkg/util/tls"
+	"github.com/Azure/ARO-RP/pkg/util/uuid"
+	utilerror "github.com/Azure/ARO-RP/test/util/error"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+type certInfo struct {
+	secretName, certSubject string
+}
+
+const (
+	managedDomainName   = "contoso.aroapp.io"
+	unmanagedDomainName = "aro.contoso.com"
+)
+
+func TestEmitCertificateExpirationStatuses(t *testing.T) {
+	expiration := time.Now().Add(time.Hour * 24 * 5)
+	expirationString := expiration.UTC().Format(time.RFC3339)
+	clusterID := uuid.DefaultGenerator.Generate()
+
+	for _, tt := range []struct {
+		name            string
+		domain          string
+		certsPresent    []certInfo
+		wantExpirations []map[string]string
+		wantWarning     []map[string]string
+		wantErr         string
+	}{
```

Comment:
The first two tests here are happy-path. There is no checking for how close a certificate is, the metric emits for all certificates. It will report 364 days until expiration for a brand new certificate with a year long life. We can use the Geneva monitors to set a threshold for alerting.